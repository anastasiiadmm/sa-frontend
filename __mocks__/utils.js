import * as reduxHooks from 'react-redux';

jest.mock('../src/redux/store');
jest.mock('react-redux');

export const mockedUseSelectors = jest.spyOn(reduxHooks, 'useSelector');
export const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch');
