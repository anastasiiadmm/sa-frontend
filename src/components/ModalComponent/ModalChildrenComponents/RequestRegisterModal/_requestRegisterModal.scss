@use 'assets/scss/_themes.scss';

.RequestRegisterModal {
  &__form-block {
    display: flex;
    gap: 13px;
    > div {
      width: 100%;
    }
  }

  &__form-modal-buttons {
    margin-top: 35px;
    display: flex;
    gap: 10px;
  }

  &__cancel-profile-button {
    background: themes.$gray-color !important;
    &:hover {
      background: rgb(112, 121, 136) !important;
    }
  }

  &__delete-profile-button {
    background: themes.$red-color !important;
    &:hover {
      background: rgba(193, 16, 48, 0.6) !important;
    }
  }

  &__close-form-button {
    background: themes.$primary-color;
  }
}
