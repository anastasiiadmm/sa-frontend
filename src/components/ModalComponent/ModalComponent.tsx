import { Divider, Modal } from 'antd';
import React from 'react';

type Props = {
  title?: string;
  open: boolean;
  closable?: boolean;
  dividerShow?: boolean;
  children?: React.ReactNode;
  handleOk?: (params: any) => any;
  handleCancel?: (params: any) => any;
  classNameTitle?: string;
};

const ModalComponent: React.FC<Props> = ({
  title,
  open,
  closable,
  dividerShow = true,
  children,
  handleOk,
  handleCancel,
  classNameTitle,
}) => {
  return (
    <Modal
      closable={closable}
      title={<span className={classNameTitle}>{title}</span>}
      open={open}
      onOk={handleOk}
      onCancel={handleCancel}
      footer={null}
    >
      {dividerShow ? <Divider /> : null}
      {children}
    </Modal>
  );
};

export default ModalComponent;
