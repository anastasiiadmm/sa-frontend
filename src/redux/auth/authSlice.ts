import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

import { IErrors, ITokens, IUser, userMutation } from 'interfaces';
import { RootState } from 'redux/hooks';
import store from 'redux/store';
import { addCookies } from 'utils/addCookies/addCookies';
import { axiosApi } from 'utils/axios-api';

interface AuthState {
  user: IUser | null;
  tokens: ITokens;
  errors: Object | null;
  commonError: IErrors | null;
  success: boolean | null;
  loading: boolean;
}

const nameSpace = 'auth';

const INITIAL_STATE = {
  user: null,
  tokens: {
    access: '',
    refresh: '',
    is_manager: false,
  },
  errors: null,
  commonError: null,
  success: null,
  loading: false,
} as AuthState;

export const loginUser = createAsyncThunk<ITokens, userMutation>(
  `${nameSpace}/loginUser`,
  async (loginData, { rejectWithValue }) => {
    try {
      const resp = await axiosApi.post('/accounts/login/', loginData);
      addCookies('refresh', resp.data.refresh);
      localStorage.setItem(
        'users',
        JSON.stringify({
          access: resp.data.access,
          is_manager: resp.data.is_manager,
        }),
      );
      return resp.data;
    } catch (e) {
      return rejectWithValue({
        detail: e?.response?.data?.detail,
        status: e?.response?.status,
      });
    }
  },
);

export const refreshToken = createAsyncThunk(`${nameSpace}/refreshToken`, async () => {
  const refresh = store.getState()?.auth?.tokens?.refresh;
  if (refresh) {
    const data = { refresh };
    const resp = await axiosApi.post('/accounts/refresh/', data);
    return resp.data;
  }
});

export const authSlice = createSlice({
  name: nameSpace,
  initialState: INITIAL_STATE,
  reducers: {
    clearAuthState: (state) => {
      state.loading = false;
      state.success = false;
      state.errors = null;
    },
    logoutUser: () => INITIAL_STATE,
    refreshAccessToken: (state, { payload }) => {
      state.tokens = { ...state.tokens, ...payload };
    },
    checkForTokens: (state, payload) => {
      state.tokens.access = payload.payload?.access;
      state.tokens.refresh = payload.payload?.refresh;
      state.tokens.is_manager = payload.payload?.is_manager;
    },
  },
  extraReducers: (builder) => {
    builder.addCase(loginUser.pending, (state) => {
      state.loading = true;
      state.success = false;
      state.commonError = null;
    });
    builder.addCase(loginUser.fulfilled, (state, { payload }) => {
      state.tokens = payload;
      state.loading = false;
      state.success = true;
      state.errors = null;
      state.commonError = null;
    });
    builder.addCase(loginUser.rejected, (state, { payload }) => {
      state.loading = false;
      state.success = false;
      if (payload && typeof payload === 'object' && 'detail' in payload && 'status' in payload) {
        state.commonError = {
          ...state.commonError,
          detail: payload.detail as string | null,
          status: payload.status as number | null,
        };
      }
    });

    builder.addCase(refreshToken.fulfilled, (state, { payload }) => {
      state.tokens = { ...state.tokens, ...payload };
    });
  },
});

export const { logoutUser, checkForTokens } = authSlice.actions;
export const authSelector = (state: RootState) => state.auth;
export default authSlice.reducer;
