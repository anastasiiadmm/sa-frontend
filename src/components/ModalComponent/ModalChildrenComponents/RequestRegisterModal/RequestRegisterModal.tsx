import { Button, Col, Form, message } from 'antd';
import bem from 'easy-bem';
import React, { useEffect, useState } from 'react';

import AlertComponent from 'components/AlertComponent/AlertComponent';
import FormField from 'components/FormField/FormField';
import 'components/ModalComponent/ModalChildrenComponents/RequestRegisterModal/_requestRegisterModal.scss';
import { getErrorMessage, removeEmptyValuesFromObject } from 'helper';
import { accountsSelector, approveFieldClimateRequest } from 'redux/accounts/accountsSlice';
import { useAppDispatch, useAppSelector } from 'redux/hooks';

type Props = {
  onClose: () => void;
};

const RequestRegisterModal: React.FC<Props> = ({ onClose }) => {
  const b = bem('RequestRegisterModal');
  const [form] = Form.useForm();
  const dispatch = useAppDispatch();
  const { inquiriesLoading: loading, inquiriesSuccess } = useAppSelector(accountsSelector);
  const [userData, setUserData] = useState({
    user: {
      last_name: '',
      first_name: '',
      middle_name: '',
      email: '',
      phone: '',
    },
    name: '',
    location: '',
  });

  useEffect(() => {
    if (userData) {
      form.setFieldsValue({
        first_name: userData?.user.first_name,
        last_name: userData?.user.last_name,
        middle_name: userData?.user.middle_name,
        email: userData?.user.email,
        phone: userData?.user.phone,
        name: userData?.name,
        location: userData?.location,
      });
    }
  }, [userData, form]);

  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;

    if (name === 'location' || name === 'name') {
      setUserData({ ...userData, [name]: value });
    } else {
      setUserData((prevUserData) => ({
        ...prevUserData,
        user: {
          ...prevUserData.user,
          [name]: value,
        },
      }));
    }
  };

  const onFinish = async () => {
    try {
      if (userData) {
        const data = removeEmptyValuesFromObject(userData);
        const registerUserObj = {
          category: 1,
          data: {
            user: {
              ...data.user,
            },
            enterprise: {
              name: data.name,
              location: data.location,
            },
          },
        };
        await dispatch(inquiriesRequests(registerUserObj)).unwrap();
        onClose();
      }
    } catch (e) {
      const errorMessage = getErrorMessage(e, 'username');
      await message.error(`${errorMessage}`);
    }
  };

  const successSend = (
    <>
      <AlertComponent
        message='Спасибо за заявку. Ваш запрос был отправлен модератору'
        type='info'
        showIcon
      />
      <Button
        className={b('close-form-button')}
        type='primary'
        style={{ width: '100%', borderRadius: 4, marginTop: 40 }}
        onClick={onClose}
      >
        Закрыть
      </Button>
    </>
  );

  return inquiriesSuccess ? (
    successSend
  ) : (
    <Col
      className={b('')}
      xs={{ span: 24, offset: 0 }}
      md={{ span: 24, offset: 0 }}
      lg={{ span: 24, offset: 0 }}
    >
      <p style={{ marginBottom: 25 }}>После отправки заявки с вами свяжется наш менеджер.</p>
      <Form
        form={form}
        initialValues={{ remember: true }}
        onFinish={onFinish}
        autoComplete='off'
        layout='vertical'
      >
        <div className={b('form-block')}>
          <FormField
            bordered
            data-testid='last_name_id'
            id='last_name_id'
            inputClassName={b('username')}
            label='Фамилия'
            name='last_name'
            placeholder='Фамилия'
            onChange={inputChangeHandler}
            rules={[{ required: true, message: 'Введите фамилию' }]}
          />

          <FormField
            bordered
            data-testid='first_name_id'
            id='first_name_id'
            inputClassName={b('username')}
            label='Имя'
            name='first_name'
            placeholder='Имя'
            onChange={inputChangeHandler}
            rules={[{ required: true, message: 'Введите имя' }]}
          />
        </div>

        <FormField
          bordered
          data-testid='middle_name_id'
          id='missle_name_id'
          inputClassName={b('username')}
          label='Отчество'
          name='middle_name'
          placeholder='Отчество'
          onChange={inputChangeHandler}
          rules={[{ required: true, message: 'Введите отчество' }]}
        />

        <div className={b('form-block')}>
          <FormField
            bordered
            data-testid='email_id_login'
            type='email'
            id='email_id'
            inputClassName={b('username')}
            label='Email'
            name='email'
            placeholder='Email'
            onChange={inputChangeHandler}
            rules={[
              {
                type: 'email',
                message: 'Введен неверный E-mail!',
              },
            ]}
          />

          <FormField
            bordered
            type='phone'
            className='username'
            name='phone'
            label='Номер телефона'
            placeholder='Номер телефона'
            onChange={inputChangeHandler}
            rules={[{ required: true, message: 'Введите номер телефона' }]}
          />
        </div>

        <FormField
          bordered
          data-testid='name_of_farm_id'
          id='name_of_farm_id'
          inputClassName={b('username')}
          label='Название колхоза/фермы/компании'
          name='name'
          placeholder='Название колхоза/фермы/компании'
          onChange={inputChangeHandler}
          rules={[{ required: true, message: 'Введите название' }]}
        />

        <FormField
          bordered
          data-testid='location_id'
          id='location_id'
          inputClassName={b('username')}
          label='Регион расположения'
          name='location'
          placeholder='Регион расположения'
          onChange={inputChangeHandler}
          rules={[{ required: true, message: 'Введите регион' }]}
        />

        <div className={b('form-modal-buttons')}>
          <Button
            type='primary'
            htmlType='submit'
            loading={loading}
            style={{ width: '100%', borderRadius: 4 }}
            className={b('login-form-button')}
          >
            Отправить запрос
          </Button>
        </div>
      </Form>
    </Col>
  );
};

export default RequestRegisterModal;
