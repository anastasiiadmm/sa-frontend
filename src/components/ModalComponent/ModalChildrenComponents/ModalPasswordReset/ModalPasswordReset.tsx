import { Button, Form } from 'antd';
import bem from 'easy-bem';
import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import AlertComponent from 'components/AlertComponent/AlertComponent';
import FormField from 'components/FormField/FormField';
import { authSelector, resetUserPasswordSendEmail } from 'redux/auth/authSlice';
import { AppDispatch } from 'redux/hooks';
import { ResetEmail } from 'types/types';
import 'containers/SignIn/_signIn.scss';

const ModalPasswordReset: React.FC = () => {
  const b = bem('SignIn');
  const [form] = Form.useForm();
  const dispatch = useDispatch<AppDispatch>();
  const { loading, success, errors } = useSelector(authSelector);
  const [state, setState] = useState<ResetEmail | null>(null);

  const onFinish = (value: ResetEmail) => {
    setState(value);
    dispatch(resetUserPasswordSendEmail(value));
  };

  const successSend = (
    <>
      <AlertComponent
        message={`Временный пароль был выслан на ${state?.email}`}
        type='info'
        showIcon
      />
      <Button
        className={b('login-form-button')}
        type='primary'
        style={{ width: '100%', borderRadius: 4 }}
      >
        Закрыть
      </Button>
    </>
  );

  const errorSend = (
    <>
      <AlertComponent message='Такой email не найден' type='error' showIcon />
      <Form
        style={{ marginTop: 20 }}
        initialValues={{ email: state?.email }}
        onFinish={onFinish}
        className='login-form'
        autoComplete='off'
        layout='vertical'
      >
        <FormField
          data-testid='email_id_login'
          type='email'
          id='email_id'
          name='email'
          label='Ваш email'
        />
        <Button
          loading={!!loading}
          className={b('login-form-button')}
          type='primary'
          htmlType='submit'
          style={{ width: '100%', borderRadius: 4 }}
        >
          Продолжить
        </Button>
      </Form>
    </>
  );

  return success ? (
    successSend
  ) : errors ? (
    errorSend
  ) : (
    <>
      <AlertComponent
        message='Для восстановления пароля к вашей учетной записи введите ваш E-mail'
        type='info'
        showIcon
      />
      <Form
        style={{ marginTop: 20 }}
        initialValues={{ remember: true }}
        onFinish={onFinish}
        className='login-form'
        autoComplete='off'
        layout='vertical'
      >
        <FormField
          data-testid='email_id_login'
          type='email'
          id='email_id'
          name='email'
          label='Ваш email'
        />
        <Button
          loading={!!loading}
          className={b('login-form-button')}
          type='primary'
          htmlType='submit'
          style={{ width: '100%', borderRadius: 4 }}
        >
          Продолжить
        </Button>
      </Form>
    </>
  );
};

export default ModalPasswordReset;
