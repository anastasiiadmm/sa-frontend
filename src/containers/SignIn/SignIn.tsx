import { Button, Col, Divider, Form, message, Row, Skeleton, Typography } from 'antd';
import { CheckboxChangeEvent } from 'antd/es/checkbox';
import bem from 'easy-bem';
import React, { lazy, Suspense, useEffect, useState } from 'react';
import { useNavigate } from 'react-router';

import logo from 'assets/images/logo.png';
import tractor from 'assets/images/tracktor.png';
import DrawerComponent from 'components/DrawerComponent/DrawerComponent';
import FormField from 'components/FormField/FormField';
import ModalComponent from 'components/ModalComponent/ModalComponent';
import useWindowWidth from 'hooks/useWindowWidth';
import { userMutation } from 'interfaces';
import { inquiriesSuccessNull } from 'redux/accounts/accountsSlice';
import { authSelector, loginUser } from 'redux/auth/authSlice';
import { useAppDispatch, useAppSelector } from 'redux/hooks';
import { userLocalStorage } from 'utils/addLocalStorage/addLocalStorage';
import 'containers/SignIn/_signIn.scss';

const RequestRegisterModal = lazy(
  () =>
    import(
      'components/ModalComponent/ModalChildrenComponents/RequestRegisterModal/RequestRegisterModal'
    ),
);
const { Title, Text } = Typography;

const SignIn: React.FC = () => {
  const b = bem('SignIn');
  const dispatch = useAppDispatch();
  const windowWidth = useWindowWidth();
  const [form] = Form.useForm();
  const { success, loading, user } = useAppSelector(authSelector);
  const history = useNavigate();
  const [checked, setChecked] = useState(true);
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [isDrawerOpen, setIsDrawerOpen] = useState(false);

  const showModal = () => {
    windowWidth <= 990 ? setIsDrawerOpen(true) : setIsModalOpen(true);
    dispatch(inquiriesSuccessNull());
  };

  const handleCancel = () => {
    windowWidth <= 990 ? setIsDrawerOpen(false) : setIsModalOpen(false);
  };

  const onFinish = async (values: userMutation) => {
    try {
      await dispatch(loginUser(values)).unwrap();
    } catch (e) {
      await message.error(e?.detail ? e?.detail : 'Ошибка авторизации');
    }
  };

  const onChangeCheckbox = (e: CheckboxChangeEvent) => {
    setChecked(e.target.checked);
  };

  const pushToMainPage = () => {
    if (success && user && userLocalStorage()) {
      history('/');
    }
  };

  useEffect(() => {
    pushToMainPage();
  }, [user, success]);

  return (
    <>
      <Row justify='space-between' data-testid='sign-in' className={b()}>
        <Col
          xs={{ span: 24, offset: 0 }}
          md={{ span: 24, offset: 0 }}
          lg={{ span: 12, offset: 0 }}
          className={b('image-block')}
        >
          <img src={tractor} alt='tractor' className={b('image-styles')} />
        </Col>
        <Col
          className={b('main-colum-styles')}
          xs={{ span: 16, offset: 5 }}
          md={{ span: 13, offset: 5 }}
          lg={{ span: 7, offset: 1 }}
        >
          <div className={b('block_form')}>
            <img src={logo} alt='logo' className={b('logo-image')} />
            <Divider style={{ margin: '20px 0' }} />
            <div className={b('form_desktop')}>
              <Text type='secondary'>Рады вас видеть</Text>
              <Title data-testid='sign_in_test' className={b('title-text')}>
                Войдите в аккаунт
              </Title>

              <Form
                form={form}
                initialValues={{ remember: true }}
                onFinish={onFinish}
                className='login-form'
                autoComplete='off'
                layout='vertical'
                size='large'
              >
                <FormField
                  bordered
                  data-testid='email_id_login'
                  type='email'
                  id='email_id'
                  name='username'
                  rules={[
                    {
                      required: true,
                      message: 'Заполните логин',
                    },
                  ]}
                  placeholder='Логин'
                  inputClassName={b('sign-in-input')}
                />

                <FormField
                  bordered
                  className={b('password_input')}
                  data-testid='password_id'
                  type='password'
                  name='password'
                  placeholder='Пароль'
                  inputClassName={b('sign-in-input')}
                />

                <FormField
                  className='checkbox-styles'
                  id='remember_id'
                  type='checkbox'
                  label={<span style={{ color: '#59647A' }}>Запомнить меня</span>}
                  valuePropName='password'
                  checked={checked}
                  onChange={onChangeCheckbox}
                />

                <Button
                  size={windowWidth <= 601 ? 'small' : 'middle'}
                  type='primary'
                  htmlType='submit'
                  loading={!!loading}
                  className={b('login-form-button')}
                >
                  Войти
                </Button>
              </Form>
              <Row className={b('buttons-row')}>
                <p style={{ margin: '17px auto' }}>Хотите стать клиентом?</p>
                <Button
                  size={windowWidth <= 601 ? 'small' : 'middle'}
                  type='primary'
                  className={b('register-button')}
                  onClick={showModal}
                >
                  Зарегистрироваться
                </Button>
              </Row>
            </div>
          </div>
        </Col>
      </Row>

      <ModalComponent
        dividerShow={false}
        title='Отправить запрос на регистрацию'
        open={isModalOpen}
        handleCancel={handleCancel}
      >
        <Suspense fallback={<Skeleton active />}>
          <RequestRegisterModal onClose={handleCancel} />
        </Suspense>
      </ModalComponent>

      <DrawerComponent
        closable
        title='Отправить запрос на регистрацию'
        width='100%'
        bodyStyle={{ padding: 20 }}
        open={isDrawerOpen}
        onClose={handleCancel}
        placement='right'
      >
        <Suspense fallback={<Skeleton active />}>
          <RequestRegisterModal onClose={handleCancel} />
        </Suspense>
      </DrawerComponent>
    </>
  );
};

export default SignIn;
