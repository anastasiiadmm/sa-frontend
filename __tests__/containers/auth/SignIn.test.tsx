import { BrowserRouter } from "react-router-dom";
import renderer, { act } from "react-test-renderer";
import { screen, render, waitFor, fireEvent } from "@testing-library/react";
import '../../../__mocks__/matchMedia.mock';
import '@testing-library/jest-dom';
import '../../../__mocks__/utils';
import { mockedDispatch, mockedUseSelectors } from "../../../__mocks__/utils";
import * as actions from '../../../src/redux/auth/authSlice';
import SignIn from "../../../src/containers/SignIn/SignIn";
import ModalComponent from "../../../src/components/ModalComponent/ModalComponent";
import ModalPasswordReset
  from "../../../src/components/ModalComponent/ModalChildrenComponents/ModalPasswordReset/ModalPasswordReset";
import RequestRegisterModal
  from "../../../src/components/ModalComponent/ModalChildrenComponents/RequestRegisterModal/RequestRegisterModal";

jest.mock('antd', () => ({
  ...jest.requireActual('antd'),
  message: {
    error: jest.fn(),
  },
}));

describe('<SignIn />', () => {
  test('Render component toMatchSnapshot()', () => {
    mockedUseSelectors.mockReturnValue([]);

    const tree = renderer
      .create(
        <BrowserRouter>
          <SignIn />
        </BrowserRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Opening a modal and closing it", () => {
    const handleClose = jest.fn();
    const yesDeleteHandler = jest.fn();

    act(() => {
      render(
        <ModalComponent open title='Восстановление пароля' handleOk={yesDeleteHandler} handleCancel={handleClose}>
          <ModalPasswordReset />
        </ModalComponent>
      );
    });
  });

  it("Opening register modal and closing", () => {
    const handleClose = jest.fn();
    const yesDeleteHandler = jest.fn();

    act(() => {
      render(
        <ModalComponent open title='Отправить запрос на регистрацию' handleOk={yesDeleteHandler} handleCancel={handleClose}>
          <RequestRegisterModal onClose={handleClose} />
        </ModalComponent>
      );
    });
  });

  test('Login form should be in the document', async () => {
    mockedUseSelectors.mockReturnValue([]);

    render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );

    const email = screen.getByPlaceholderText('Логин') as HTMLInputElement;
    await waitFor(() => {
      expect(email).toBeInTheDocument();
    });
  });

  test('Should success on login', async () => {
    mockedUseSelectors.mockReturnValue([]);

    const dispatch = jest.fn();
    mockedDispatch.mockReturnValue(dispatch);

    const mockedLoginUserComplete = jest.spyOn(actions, 'loginUser');

    render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );

    expect(screen.getByTestId('sign-in')).toBeInTheDocument();
    const emailInput = screen.getByPlaceholderText('Логин');
    const passwordInput = screen.getByPlaceholderText('Пароль');
    const button = await screen.findByRole('button', { name: 'Войти' });
    await act(async () => {
      fireEvent.change(emailInput, { target: { value: 'manager' } });
      fireEvent.change(passwordInput, { target: { value: 'P33f2kmu!' } });
      fireEvent.click(button);
    });

    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(mockedLoginUserComplete).toHaveBeenCalledWith({
      username: 'manager',
      password: 'P33f2kmu!',
    });
  });

  test('Register form should change success', async () => {
    const handleClose = jest.fn();
    const yesDeleteHandler = jest.fn();

    render(
      <ModalComponent open title='Отправить запрос на регистрацию' handleOk={yesDeleteHandler} handleCancel={handleClose}>
        <RequestRegisterModal onClose={handleClose}/>
      </ModalComponent>
    );

    await act(async () => {
      const lastNameInput = screen.getByLabelText('Фамилия');
      const firstNameInput = screen.getByLabelText('Имя');
      const middleNameInput = screen.getByLabelText('Отчество');
      const emailInput = screen.getByLabelText('Email');
      const phoneInput = screen.getByLabelText('Номер телефона');
      const nameInput = screen.getByLabelText('Название колхоза/фермы/компании');
      const locationInput = screen.getByLabelText("Регион расположения");
      const button = await screen.findByRole('button', { name: 'Отправить запрос' });

      await act(async () => {
        fireEvent.change(lastNameInput, { target: { value: 'Test last name' } });
        expect(lastNameInput).toHaveDisplayValue('Test last name');
        fireEvent.change(firstNameInput, { target: { value: 'Test first name' } });
        expect(firstNameInput).toHaveDisplayValue('Test first name');
        fireEvent.change(middleNameInput, { target: { value: 'Test middle name' } });
        expect(middleNameInput).toHaveDisplayValue('Test middle name');
        fireEvent.change(emailInput, { target: { value: 'email@gmail.com' } });
        expect(emailInput).toHaveDisplayValue('email@gmail.com');
        fireEvent.change(phoneInput, { target: { value: '+7 (999) 999-99-99' } });
        expect(phoneInput).toHaveDisplayValue('+7 (999) 999-99-99');
        fireEvent.change(locationInput, { target: { value: 'Test location' } });
        expect(locationInput).toHaveDisplayValue('Test location');
        fireEvent.change(nameInput, { target: { value: 'Test name' } });
        expect(nameInput).toHaveDisplayValue('Test name');

        fireEvent.click(button);
      })
    });
  });

  test('Should show success message for successful login', async () => {
    mockedUseSelectors.mockReturnValue([]);

    const dispatch = jest.fn();
    mockedDispatch.mockReturnValue(dispatch);

    render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );

    expect(screen.getByTestId('sign-in')).toBeInTheDocument();
    const emailInput = screen.getByPlaceholderText('Логин');
    const passwordInput = screen.getByPlaceholderText('Пароль');
    const button = await screen.findByRole('button', { name: 'Войти' });
    await act(async () => {
      fireEvent.change(emailInput, { target: { value: 'manager' } });
      fireEvent.change(passwordInput, { target: { value: 'reto4321' } });
      fireEvent.click(button);
    });

    await waitFor(() => {
      expect(window.location.pathname).toBe('/');
    });
  });


});
