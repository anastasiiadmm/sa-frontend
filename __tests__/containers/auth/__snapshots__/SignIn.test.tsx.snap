// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> Render component toMatchSnapshot() 1`] = `
<div
  className="ant-row ant-row-space-between SignIn css-dev-only-do-not-override-mxhywb"
  data-testid="sign-in"
  style={{}}
>
  <div
    className="ant-col ant-col-xs-24 ant-col-xs-offset-0 ant-col-md-24 ant-col-md-offset-0 ant-col-lg-12 ant-col-lg-offset-0 css-dev-only-do-not-override-mxhywb"
    style={{}}
  >
    <img
      alt="tractor"
      className="SignIn__image-styles"
      src={
        {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    className="ant-col SignIn__main-colum-styles ant-col-xs-13 ant-col-xs-offset-5 ant-col-md-13 ant-col-md-offset-5 ant-col-lg-7 ant-col-lg-offset-1 css-dev-only-do-not-override-mxhywb"
    style={{}}
  >
    <img
      alt="logo"
      src={
        {
          "process": [Function],
        }
      }
    />
    <p
      className="SignIn__subtitle"
    >
      Добро пожаловать
    </p>
    <h1
      className="ant-typography SignIn__title css-dev-only-do-not-override-mxhywb"
      data-testid="sign_in_test"
      style={
        {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Вход в аккаунт
    </h1>
    <form
      autoComplete="off"
      className="ant-form ant-form-vertical css-dev-only-do-not-override-mxhywb login-form"
      id="register"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="ant-form-item css-dev-only-do-not-override-mxhywb"
      >
        <div
          className="ant-row ant-form-item-row css-dev-only-do-not-override-mxhywb"
          style={{}}
        >
          <div
            className="ant-col ant-form-item-label css-dev-only-do-not-override-mxhywb"
            style={{}}
          >
            <label
              className=""
              htmlFor="register_username"
              title="Логин"
            >
              Логин
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control css-dev-only-do-not-override-mxhywb"
            style={{}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <input
                  className="ant-input css-dev-only-do-not-override-mxhywb input-styles undefined"
                  disabled={false}
                  id="register_username"
                  name="username"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Логин"
                  readOnly={false}
                  style={{}}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-form-item PasswordField css-dev-only-do-not-override-mxhywb"
      >
        <div
          className="ant-row ant-form-item-row css-dev-only-do-not-override-mxhywb"
          style={{}}
        >
          <div
            className="ant-col ant-form-item-label css-dev-only-do-not-override-mxhywb"
            style={{}}
          >
            <label
              className="ant-form-item-required"
              htmlFor="register_password"
              title="Пароль"
            >
              Пароль
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control css-dev-only-do-not-override-mxhywb"
            style={{}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <span
                  className="ant-input-affix-wrapper ant-input-password input-styles undefined ant-input-affix-wrapper-has-feedback css-dev-only-do-not-override-mxhywb"
                  onClick={[Function]}
                  style={
                    {
                      "borderRadius": "7px",
                    }
                  }
                >
                  <input
                    aria-required="true"
                    autoComplete="on"
                    className="ant-input css-dev-only-do-not-override-mxhywb"
                    disabled={false}
                    hidden={null}
                    id="register_password"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Пароль"
                    readOnly={false}
                    style={null}
                    type="password"
                    value=""
                  />
                  <span
                    className="ant-input-suffix"
                  >
                    <span
                      aria-label="eye-invisible"
                      className="anticon anticon-eye-invisible ant-input-password-icon"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      role="img"
                      tabIndex={-1}
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="eye-invisible"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                        />
                        <path
                          d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                        />
                      </svg>
                    </span>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-form-item css-dev-only-do-not-override-mxhywb"
      >
        <div
          className="ant-row ant-form-item-row css-dev-only-do-not-override-mxhywb"
          style={{}}
        >
          <div
            className="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-mxhywb"
            style={{}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <label
                  className="ant-checkbox-wrapper ant-checkbox-wrapper-checked ant-checkbox-wrapper-in-form-item css-dev-only-do-not-override-mxhywb"
                >
                  <span
                    className="ant-checkbox css-dev-only-do-not-override-mxhywb ant-checkbox-checked"
                    style={{}}
                  >
                    <input
                      checked={true}
                      className="ant-checkbox-input"
                      disabled={false}
                      id="remember_id"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyPress={[Function]}
                      onKeyUp={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="ant-checkbox-inner"
                    />
                  </span>
                  <span>
                    Запомнить меня
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-primary SignIn__login-form-button"
        disabled={false}
        onClick={[Function]}
        style={
          {
            "borderRadius": 4,
            "width": "100%",
          }
        }
        type="submit"
      >
        <span>
          Продолжить
        </span>
      </button>
    </form>
    <div
      className="ant-row SignIn__buttons-row css-dev-only-do-not-override-mxhywb"
      style={{}}
    >
      <p>
        Хотите стать клиентом?
      </p>
      <button
        className="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          Зарегистрироваться
        </span>
      </button>
    </div>
  </div>
</div>
`;
